asv_bench/benchmarks/groupby.py

```diff
@@ -608,22 +608,29 @@ class Cumulative:
 
     def setup(self, dtype, method):
         N = 500_000
+        keys = np.random.randint(0, 100, size=N)
         vals = np.random.randint(-10, 10, (N, 5))
         null_vals = vals.astype(float, copy=True)
         null_vals[::2, :] = np.nan
         null_vals[::3, :] = np.nan
-        df = DataFrame(vals, columns=list("abcde"), dtype=dtype)
-        null_df = DataFrame(null_vals, columns=list("abcde"), dtype=dtype)
-        keys = np.random.randint(0, 100, size=N)
+
+        df = DataFrame(vals, columns=list("abcde")).astype(dtype, copy=False)
         df["key"] = keys
-        null_df["key"] = keys
         self.df = df
-        self.null_df = null_df
+
+        if dtype != "int64":
+            # Would raise on DataFrame construction with int64
+            null_df = DataFrame(null_vals, columns=list("abcde"), dtype=dtype)
+            null_df["key"] = keys
+            self.null_df = null_df
 
     def time_frame_transform(self, dtype, method):
         self.df.groupby("key").transform(method)
 
     def time_frame_transform_many_nulls(self, dtype, method):
+        if dtype == "int64":
```

Comment:
Hmm, this may cause the benchmark tracking to look odd with the prior benchmark having some finite time to then completing almost instantly.

Could this benchmark be wholly refactored like:

```
    param_names = ["dtype", "method", "with_nans"]
    params = [
        ["float64", "int64", "Float64", "Int64"],
        ["cummin", "cummax", "cumsum"],
        [True, False]
    ]

    def setup(self, dtype, method, with_nans):
          vals = np.random.randint(-10, 10, (N, 5))
          if with_nans
              if dtype = "int64":
                  raise NotImplementedError
              else:
                  vals = vals.astype(float, copy=True)
                  vals[::2, :] = np.nan
                  vals[::3, :] = np.nan
          ....

    def time_frame_transform(self, dtype, method, with_nans):
```

Reply:
sure